### ISSUE TITLE
Decide input/output mode based on whether stdin is interactive.

### ISSUE THREAD

**Issue by Janfel** [astrand/xclip#115]
Currently, calling `xclip` without any arguments always defaults to `-in` mode. This means the user needs to use it like this:

```sh
xclip -o # Prints the contents of the clipboard.
foo | xclip
xclip -o | bar
```

But as an interactive user, I would much rather just be able to use it like this:

```sh
xclip # Prints the contents of the clipboard.
foo | xclip
xclip | bar
# Or even like tee(1).
foo | xclip | bar
```

I believe that writing the contents of the clipboard to the console is a much more common use case than writing text to be sent to the clipboard. Thus, it should be the default when calling `xclip` without any arguments.

The first three lines could be accomplished by defaulting to `isatty(stdin) ? "-out" : "-in"` instead of always choosing `-in`. This should only break scripts that rely on the already uncommon use case described above to be the default, which I assume to be very rare.

The fourth line would require a third `-io` or `-inout` mode, that writes `stdin` to the clipboard as well as piping it to `stdout`. This mode should not wait for `stdin` to reach `EOF` before printing to `stdout`. If such a mode were to be implemented, it should be the default in case `stdin` is not interactive. The general default value should then be `isatty(stdin) ? "-out" : "-inout"`, else the fourth line would not work.

I believe that this would be a big improvement to usability, intuitiveness and general user experience. I’m looking forward to hearing your opinion on this.

**Comment by Janfel** [comment]
It seems the main part of this issue has already been resolved in #65. I hope there will soon be a release that incorporates this change. I will leave this open in case the idea of an `-inout` mode is interesting to you. Feel free to close it if that is not the case.

**Comment by hackerb9** [comment]
I agree with this suggestion. Making the most common case the easiest makes sense to me.  If a script wants to ensure a specific mode instead of relying on autodetection, it could always specify -in/-out/-filter.

However, the maintainer of xclip is not yet convinced that using isatty() a good idea. Feel free to weigh in on #88 if you have any helpful thoughts.
